### PR TITLE
Add ability to control clock nemesis targets.

### PR DESCRIPTION
**Pull request by stevana**
Most of the work here was done by @Danten.

### CODE REVIEW

**Comment by stevana** [comment]
Like this?

File: jepsen/src/jepsen/nemesis/time.clj [hunk 1]
```diff
@@ -138,35 +138,50 @@
     (teardown! [_ test]
       (reset-time! test))))
 
-(defn reset-gen
-  "Randomized reset generator. Performs resets on random subsets of the tests'
+(defn reset-gen-select
```
Comment (aphyr) [inline jepsen/src/jepsen/nemesis/time.clj:141]: These aren't generators any more; they're functions which return generators, and their use pattern is going to be different. Might want to make that explicit in the docstrings.

File: jepsen/src/jepsen/nemesis/time.clj [hunk 2]
```diff
@@ -138,35 +138,50 @@
...
-   :f     :bump
-   :value (zipmap (util/random-nonempty-subset (:nodes test))
-                  (repeatedly (fn []
-                                (long (* (rand-nth [-1 1])
-                                         (Math/pow 2 (+ 2 (rand 16))))))))})
-
-(defn strobe-gen
-  "Randomized clock strobe generator. On random subsets of the test's nodes,
+  [select]
+  (fn [test process]
```
Comment (aphyr) [inline jepsen/src/jepsen/nemesis/time.clj:145]: You could also accomplish this by adding a single-arity form to bump-gen, strobe-gen, etc.

File: jepsen/src/jepsen/nemesis/combined.clj [hunk 3]
```diff
@@ -234,12 +234,20 @@
...
-                                   :strobe         :strobe-clock
-                                   :bump           :bump-clock})
-                       (gen/delay (:interval opts default-interval)))]
+          db (:db opts)
+          target-specs (:targets (:clock opts) (node-specs db))
+          targets (fn [test] (db-nodes test db
+                                       (some-> target-specs seq rand-nth)))
+          clock-gen (gen/phases
+                     (gen/once {:type :info, :f :check-offsets})
+                     (gen/mix ((juxt reset-gen-select bump-gen-select strobe-gen-select)
```
Comment (aphyr) [inline jepsen/src/jepsen/nemesis/combined.clj:243]: This took me about 15 minutes to figure out, haha. I thought that -select was somehow acting as a verb, but it's actually a function which yields generators. I think I get it now. Comments might be helpful.

File: jepsen/src/jepsen/nemesis/time.clj (hunk 1 above)
Comment (stevana) [reply jepsen/src/jepsen/nemesis/time.clj:141]: Fixed.

File: jepsen/src/jepsen/nemesis/combined.clj (hunk 3 above)
Comment (stevana) [reply jepsen/src/jepsen/nemesis/combined.clj]: Simplified and added comment.

File: jepsen/src/jepsen/nemesis/time.clj (hunk 2 above)
Comment (stevana) [reply jepsen/src/jepsen/nemesis/time.clj:146]: I thought about doing this, but I figured a separate function would be more clear.

**Comment by aphyr** [comment]
All good, thanks!

**Comment by aphyr** [comment]
Uh, this... doesn't actually compile. Reverting for now.

**Pull request merged**